extensions/kafka/src/main/java/com/hazelcast/jet/kafka/TopicsConfig.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.kafka;
+
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.hazelcast.internal.util.Preconditions.checkNotNull;
+import static java.util.Collections.unmodifiableMap;
+
+/**
+ * Contains the configuration for all Kafka topics that will be consumed by the
+ * {@linkplain KafkaProcessors#streamKafkaP Kafka source processor}.
+ */
+public class TopicsConfig implements Serializable {
+
+    private static final ILogger LOGGER = Logger.getLogger(TopicsConfig.class);
+    private static final long serialVersionUID = 1L;
+
+    private final Map<String, TopicConfig> topicConfigs = new HashMap<>();
```

Comment:
we want a deterministic order of topics, it would be easier to use `LinkedHashMap`

Reply:
https://github.com/hazelcast/hazelcast/pull/21546/commits/01625a03e87e13790d0cc4a73a00579b21786fb4